src/tests/conftest.py

```diff
@@ -31,7 +18,69 @@
 
 
 def pytest_configure(config: Config) -> None:
-    pytest.dbs_dir = Path(__file__).parent / 'databases'
+    pytest.dbs_dir = Path(__file__).parent / "databases"
+
+
+@pytest.fixture(scope="module")
+def test_dbs(request):
+    """
+    Test database fixture using SQLite databases.
+
+    Uses pre-converted .db files and creates temporary copies for test isolation.
+    Changes made during tests won't affect the original .db files.
+    """
+    db_configs = request.param if hasattr(request, "param") else []
+    test_databases = {}
+    temp_resources = []  # Track resources for cleanup
+
+    for db_config in db_configs:
+        src_path = db_config["src"]
+        db_name = src_path.name
+
+        # Use pre-converted SQLite .db files with temporary copies for isolation
+        sqlite_file = src_path.parent / f"{db_name}.db"
+
+        if not sqlite_file.exists():
+            raise FileNotFoundError(
+                f"SQLite database not found: {sqlite_file}\n"
+                f"Please convert it first using your conversion script."
+            )
```

Comment:
The pythonic approach would be to assume the file exists, and wrap the `shutil.copy2()` call around a try-catch block:
```suggestion
        try:
            shutil.copy2(sqlite_file, temp_db_file)
        except FileNotFoundError as exc:
            raise FileNotFoundError(
                f"SQLite database not found: {sqlite_file}\n"
                f"Please convert it first using your conversion script."
            ) from exc
```
_Note:_ not 100% sure the exception to catch is `FileNotFoundError`, but based on `shutil.copy2()` code, looks that way.